doc/use_cases.md

```diff
@@ -0,0 +1,49 @@
+
+For this use case we would have a smaller set of posteriors as the requirements are very strict. Initial set comes from https://github.com/stan-dev/stat_comp_benchmarks. This set will be gradually expanded.
+
+In this use case we assume an algorithm should work well for all posteriors in the strict testing set.
+
+This kind of strict testing could be run every time before new release and every time after any algorithm changes in a development branch.
+
+## Code speed testing
+
+If the above strict test is passed, the same test can be used for wall clock timing comparisons of different code versions run on same hardware and OS.
+
+Even if the algorithm doesn't change, various changes in, for example, Stan math library, autodiff, parallelization implementation, and compiler options can affect the wall clock time. The small test set can be used to test code speed by running earlier and new version in the same computer and OS. See example in https://github.com/stan-dev/performance-tests-cmdstan
+
+## Efficiency comparisons of inference algorithms with asymptotically decreasing bias and variance
+
+Assuming we have different inference algorithms, small variants of some specific algorithm, for example, having different warm-up adaptations in dynamic HMC, and the implementations of these have passed the above strict checking. We are interested in testing efficiency of these algorithms with more complex posteriors. Now to make the set of more interesting models have enough variety, we assume that careful use dynamic HMC, convergence diagnostics, and SBC will provide the reference posterior. We don't claim these reference posteriors would be 100% correct, and if your new algorithm produces different result there is reason to check which result is more likely to closer to true posterior. In efficiency comparisons we assume the algorithms are working, that is, they have passed the previous more strict tests with smaller set of models. For a bigger set of models we don't make as strict tests, but it is useful to make some checks that produced  posterior is close to the reference posterior. The main interest in this use case is however in inference diagnostics and efficiency measures. Inference diagnostics should detect if the tested inference algorithm is not able to produce accurate results. An inference algorithm can be useful if it works fast for some set of models and it can be diagnosed when it doesn't work. Useful efficiency measures are effective sample size divided by the number of log-density and gradient evaluations and effective sample size per second. The first one is useful when comparing high-level algorithms as it doesn't care how well log-density and gradient evaluations are optimized and the second one can be useful when comparing algorithms where the choice of algorithm can affect the computational cost of log-density and gradient evaluation (e.g. use of Laplace integration over the latent values)-
```

Comment:
I think in general we should try to keep use cases focused on being use cases, i.e. show how the database is intended to be used. Now there is also a mashup with how we should do the efficiency testing. I think we can have some recommendations on this either as a second or third paragraph, or do separate documents describing how to do efficiency testing using posteriordb.

The risk now is that people that want to read about what use cases need to read about how we should do efficacy testing and people that want to know how to do efficacy testing wouldn't read the use cases.

Reply:
This is based on the discourse discussion. If we are two vague in the use cases, people don't understand the requirements. There will be a different more explicit document for efficiency testing, but since it doesn't exist yet we can't refer to it.

Reply:
True.